bootstrap command resulting in panic

Hi! I'm new to Goa (I heard Brian's talk at [Abstractions](abstractions.io)).

I'm getting a nil pointer error when running `bootstrap`. There's no doubt in my mind that my spec is incorrect in some way; I just image that throwing a nil pointer error is also probably not the intended behavior of the Goa tool.

Before I show you the logs and the files, I wanted to say what I did to unearth my solution.

First, note that giving an attribute to the `Checkstyle` and the `Findbugs` type fixes the problem.
Second, when I remove those members and return them to empty types, if I change the `BugProfile` type so that it doesn't contain the nested `Checkstyle` and `Findbugs` types, I don't get an nil pointer and `goagen` gives me a pleasant and readable error (`type "Checkstyle": attribute type is nil`). So it would appear that Goa panics when encountering a nested type which is empty.

Here's the panic log:

```
goagen bootstrap -o src/ -d github.com/alligrader/gradebook-api/designs
exit status 2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0xa4268]

goroutine 1 [running]:
panic(0x39b920, 0xc4200140d0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/goadesign/goa/design.(*UserTypeDefinition).ToObject(0xc42017d260, 0xc42017bb00)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/types.go:657 +0x28
github.com/goadesign/goa/design.(*AttributeDefinition).Validate(0xc4201da180, 0xc42017d5c0, 0x13, 0x5f5f00, 0xc42017d2a0, 0xc42017bbd0)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/validation.go:463 +0x1dc
github.com/goadesign/goa/design.(*AttributeDefinition).Validate(0xc4201da080, 0xc42017d5a0, 0x11, 0x5f5f00, 0xc42017d2a0, 0xc4201b9e30)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/validation.go:479 +0x451
github.com/goadesign/goa/design.(*UserTypeDefinition).Validate(0xc42017d2a0, 0x414e9b, 0xe, 0x5f5c40, 0xc4201c8460, 0x0)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/validation.go:520 +0xa8
github.com/goadesign/goa/design.(*ActionDefinition).Validate(0xc4201c8460, 0xc4201b9830)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/validation.go:333 +0x886
github.com/goadesign/goa/design.(*ResourceDefinition).validateActions(0xc42019e380, 0xc4201b9dd0)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/validation.go:231 +0xe2
github.com/goadesign/goa/design.(*ResourceDefinition).Validate(0xc42019e380, 0xc42017d580)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/validation.go:209 +0x6f
github.com/goadesign/goa/design.(*APIDefinition).Validate.func1(0xc42019e380, 0xc4201b9650, 0x410496)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/validation.go:91 +0x55
github.com/goadesign/goa/design.(*APIDefinition).IterateResources(0xc420196580, 0xc420209a80, 0xc420209b40, 0x18ea14)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/definitions.go:607 +0x1d8
github.com/goadesign/goa/design.(*APIDefinition).Validate(0xc420196580, 0x5f5c00, 0xc420196580)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/validation.go:118 +0x232
github.com/goadesign/goa/dslengine.validateSet(0xc42017bb80, 0x1, 0x1, 0x635548, 0x0)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/dslengine/runner.go:312 +0x120
github.com/goadesign/goa/design.(*APIDefinition).IterateSets(0xc420196580, 0x453700)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/design/definitions.go:458 +0x96
github.com/goadesign/goa/dslengine.Run(0x5f7580, 0xc42017d2c0)
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/goadesign/goa/dslengine/runner.go:108 +0x197
main.main()
        /Users/robbiemckinstry/workspace/go-workspace/src/github.com/alligrader/gradebook-api/goagen024301358/main.go:24 +0x63
```

Here's the content of `designs/api.go`

```
package designs

import (
    . "github.com/goadesign/goa/design"
    . "github.com/goadesign/goa/design/apidsl"
)

var _ = API("Gradebook API", func() { // "My API" is the name of the API used in docs
    Title("Gradebook")                // Documentation title
    Description("The next big thing") // Longer documentation description
    Host("localhost:8081")            // Host used by Swagger and clients
    Scheme("http")                    // HTTP scheme used by Swagger and clients
    BasePath("/api")                  // Base path to all API endpoints
    Consumes("application/json")      // Media types supported by the API
    Produces("application/json")      // Media types generated by the API
})
```

Then, both my FindBug s and my Checkstyle types are in their own files, which have no membes, just a description:

`designs/findbugs.go`:
```
package designs

import (
    _ "github.com/goadesign/goa/design"
    . "github.com/goadesign/goa/design/apidsl"
)

// FindBugs represents all configurables for the FindBugs static checker
var FindBugs = Type("Findbugs", func() {
    Description("Description of which Findbugs bugs are checked.")

    // TODO fill in with the list of FindBugs elements
})
```

`designs/checkstyle.go`:
```
package designs

import (
    _ "github.com/goadesign/goa/design"
    . "github.com/goadesign/goa/design/apidsl"
)

// Checkstyle contains a full specification of which Checkstyle lints are to be checked.
var Checkstyle = Type("Checkstyle", func() {
    Description("Description of which Checkstyle lints are checked.")

    // TODO fill in with the list of Checkstyle attributes
})
```

Anyway, that you for developing Goa. I'm very thankful for your contribution to FOSS and especially to the Go community. Much love.

--- a/design/apidsl/type.go
+++ b/design/apidsl/type.go
@@ -46,6 +46,8 @@ func Type(name string, dsl func()) *design.UserTypeDefinition {
 	}
 	if dsl == nil {
 		t.Type = design.String
+	} else {
+		t.Type = make(design.Object)
 	}
 	design.Design.Types[name] = t
 	return t

--- a/dslengine/runner_test.go
+++ b/dslengine/runner_test.go
@@ -143,4 +143,25 @@ var _ = Describe("DSL errors", func() {
 			Ω(dslengine.Errors[0].Line).Should(Equal(lineNumber))
 		})
 	})
+
+	Context("with DSL using an empty type", func() {
+		BeforeEach(func() {
+			API("foo", func() {})
+			Resource("bar", func() {
+				Action("baz", func() {
+					Payload("use-empty")
+				})
+			})
+			Type("use-empty", func() {
+				Attribute("e", "empty")
+			})
+			Type("empty", func() {
+			})
+			dslengine.Run()
+		})
+
+		It("does not panic", func() {
+			Ω(func() { dslengine.Run() }).ShouldNot(Panic())
+		})
+	})
 })

